src/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -741,7 +741,15 @@ public override bool Equals(object comparand)
 
         public override int GetHashCode()
         {
-            return (Name + "=" + Value + ";" + Path + "; " + Domain + "; " + Version).GetHashCode();
+            // Copied from Tuple.CombineHashCodes
+            int hash1 = Name.GetHashCode();
+            hash1 = ((hash1 << 5) + hash1) ^ Value.GetHashCode();
+            
+            int hash2 = Path.GetHashCode();
+            hash2 = ((hash2 << 5) + hash2) ^ Domain.GetHashCode();
+            
+            int hash3 = ((hash1 << 5) + hash1) ^ hash2;
+            return ((hash3 << 5) + hash3) ^ Version; // Version is already an int, no need to call GetHashCode
```

Comment:
This could be simplified with a helper:
```C#
public override int GetHashCode() => 
    CombineHashCodes(
        CombineHashCodes(
            CombineHashCodes(Name.GetHashCode, Value.GetHashCode(), 
            CombineHashCodes(Path.GetHashCode, Domain.GetHashCode)), 
        Version));

private static int CombineHashCode(int hash1, int hash2) => ((hash1 << 5) + hash1) ^ hash2;
```